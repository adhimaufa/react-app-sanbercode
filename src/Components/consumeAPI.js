import React,{Component} from 'react';
import '../App.css';

export default class ConsumeAPI extends Component{
    constructor(props){
        super(props)
        this.state={
            items : [],
            isLoading : true
        }
    }

    coomponentDidMount(){
        fetch("http://newsapi.org/v2/everything?q=apple&from=2020-07-02&to=2020-07-02&sortBy=popularity&apiKey=9166bcc7e93b48d09be0f264ac02dfbd")
            .then(response => response.json())
            .then(data => this.setState({items:data,isLoading:false}))
    }

    render(){
        const {items,isLoading} = this.state
        if(isLoading){
            return(
            <div className="Content">
                <p>Loading.....</p>
            </div>);
        }
        return(
            <div className="Content">   
                <ul>
                    {items.map((item,index) => <li key={index}>{item.author}</li>)}
                </ul>
            </div>
        );
    }
}