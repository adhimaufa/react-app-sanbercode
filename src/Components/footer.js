import React from "react";
import '../App.css';

export default function Footer() {
    return( 
        <div className="Footer">
            <h3>Let's be friend</h3>
            <div className="social-icon">
				<a href="https://twitter.com/agungtpras" target="_blank" rel="noopener noreferrer">Twitter</a>
				<a href="https://twitter.com/agungtpras" target="_blank" rel="noopener noreferrer">Telegram</a>
				<a href="https://twitter.com/agungtpras" target="_blank" rel="noopener noreferrer">Linked In</a>
				<a href="https://twitter.com/agungtpras" target="_blank" rel="noopener noreferrer">Github</a>
				<a href="https://twitter.com/agungtpras" target="_blank" rel="noopener noreferrer">Dribbble</a>
			</div>
            <small><a href="https://twitter.com/agungtpras" target="_blank" rel="noopener noreferrer">auf@tuta.io</a> Â© Auf. All Rights Reserved.</small>
        </div>
    );
  }