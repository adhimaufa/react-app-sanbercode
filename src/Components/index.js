import Home from './pageHome';
import Konversi from './pageLiftingSU';
import Hooks  from './pageHooks';
import Reduxs from '../Components/pageRedux';
import TodoList from './pageTodoList'
import ConsumeAPI from './consumeAPI'
import Footer from './footer'

export {Home,Konversi,Hooks,Reduxs,TodoList,ConsumeAPI,Footer};