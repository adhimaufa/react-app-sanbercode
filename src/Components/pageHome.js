import React from "react";
import '../App.css';

export default function Home() {
    return( 
        <div className="Content">
            {/* eslint-disable-next-line */}
            <h2>Hi, I'm Agung Tri Prasetyo ðŸ‘‹</h2>
            <p> I do big apologize to you Mr. Arjun, I still know very little and still don't understand much much much much moreeee about react.
                Actually I could not find an idea. So to fulfill points in the rules of the project, I summarized what I had done during bootcamp on this website project.
            </p>
        </div>
    );
  }