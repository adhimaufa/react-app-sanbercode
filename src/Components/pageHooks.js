import React, { useState, useEffect } from 'react';
import logo from '../logo.svg'
export default function Hooks() {
  const [count, setCount] = useState(0);

//   didupdate didmount
  useEffect(() => {
    document.title = `You clicked ${count} times`;
  },[count])

//   unmount 
  useEffect(() => {
    console.log('Unmounting');
  },[]);

  return (
    <div className="App App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <p>You Clicked {count} Times</p>
      <button className="button-style" onClick={() => setCount(count + 1)}>Add</button>
      <button className="button-style"onClick={() => setCount(0)}>Clear</button>
    </div>
  );
}