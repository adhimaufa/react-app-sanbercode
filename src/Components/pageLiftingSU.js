import React,{Component} from 'react';
import logo from '../logo.svg';
import '../App.css';


// =========Nomor 1 Detik - Menit===========

const scaleNames = {
  s: 'Second',
  m: 'Minute'
};

function toSecond(minute) {
  return minute*60;
}

function toMinute(second) {
  return second/60;
}

function tryConvert(time, convert) {
  const input = parseFloat(time);
  if (Number.isNaN(input)) {
    return '';
  }
  const output = convert(input);
  const rounded = Math.round(output * 1000) / 1000;
  return rounded.toString();
}

class TimeInput extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.props.onTimeChange(event.target.value);
  }

  render() {
    const time = this.props.time;
    const scale = this.props.scale;
    return (
      <fieldset>
        <legend>Enter time in {scaleNames[scale]}:</legend>
        <input value={time}
               onChange={this.handleChange} />
      </fieldset>
    );
  }
}

class CalculatorTime extends Component {
  constructor(props) {
    super(props);
    this.handleSecondChange = this.handleSecondChange.bind(this);
    this.handleMinuteChange = this.handleMinuteChange.bind(this);
    this.state = {time: '', scale: 's'};
  }

  handleSecondChange(time) {
    this.setState({scale: 's', time});
  }

  handleMinuteChange(time) {
    this.setState({scale: 'm', time});
  }

  render() {
    const scale = this.state.scale;
    const time = this.state.time;
    const second = scale === 'm' ? tryConvert(time, toSecond) : time;
    const minute = scale === 's' ? tryConvert(time, toMinute) : time;

    return (
      <div>
        <fieldset>
          <legend>Second to Minute Converter</legend>
          <TimeInput
            scale="s"
            time={second}
            onTimeChange={this.handleSecondChange} />
          <TimeInput
            scale="m"
            time={minute}
            onTimeChange={this.handleMinuteChange} />
        </fieldset>
      </div>
    );
  }
}

// =========Nomor 2 Gram - Kilogram===========
const scaleName = {
  gr: 'Gram',
  kg: 'Kilogram'
};

function toGram(kilogram) {
  return kilogram*1000;
}

function toKilogram(gram) {
  return gram/1000;
}

function tryConverts(weight, convert) {
  const input = parseFloat(weight);
  if (Number.isNaN(input)) {
    return '';
  }
  const output = convert(input);
  const rounded = Math.round(output * 1000) / 1000;
  return rounded.toString();
}

class WeightInput extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.props.onWeightChange(event.target.value);
  }

  render() {
    const weight = this.props.weight;
    const scale = this.props.scale;
    return (
      <fieldset>
        <legend>Enter time in {scaleName[scale]}:</legend>
        <input value={weight}
               onChange={this.handleChange} />
      </fieldset>
    );
  }
}

export class CalculatorWeight extends Component {
  constructor(props) {
    super(props);
    this.handleGramChange = this.handleGramChange.bind(this);
    this.handleKilogramChange = this.handleKilogramChange.bind(this);
    this.state = {weight: '', scale: 'gr'};
  }

  handleGramChange(weight) {
    this.setState({scale: 'gr', weight});
  }

  handleKilogramChange(weight) {
    this.setState({scale: 'kg', weight});
  }

  render() {
    const scale = this.state.scale;
    const weight = this.state.weight;
    const gram = scale === 'kg' ? tryConverts(weight, toGram) : weight;
    const kilogram = scale === 'gr' ? tryConverts(weight, toKilogram) : weight;

    return (
      <div>
        <fieldset>
        <legend>Gram to Kilogram Converter</legend>
          <WeightInput
            scale="gr"
            weight={gram}
            onWeightChange={this.handleGramChange} />
          <WeightInput
            scale="kg"
            weight={kilogram}
            onWeightChange={this.handleKilogramChange} />
        </fieldset>
      </div>
    );
  }
}

export default function Konversi() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <div className="display">
          <CalculatorTime/>
          <CalculatorWeight/>
        </div>
      </header>
    </div>
  );
}
