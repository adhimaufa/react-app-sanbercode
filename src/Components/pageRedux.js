import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { incrementAsync, decrementAsync } from '../Actions';
import '../App.css';
import logo from '../logo.svg'

function Reduxs() {
  const dispatch = useDispatch();
  const counter = useSelector(state => state.counter);

  const _add = () => {
    dispatch(incrementAsync());
  }

  const _substract = () => {
    dispatch(decrementAsync());
  }

  return (
    <div className="App App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <div>Counter: {counter}</div>
      <div style={{ display: 'flex', flexDirection: 'row' }}>
        <button className="button-style" style={{ width: 100 }} onClick={_substract}>-</button>
        <button className="button-style" style={{ width: 100 }} onClick={_add}>+</button>
      </div>
    </div>
  );
}

export default Reduxs;