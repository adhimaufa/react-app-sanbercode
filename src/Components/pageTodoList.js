import React,{Component} from 'react';
import logo from '../logo.svg';
import '../App.css';

export default class TodoList extends Component{
    state = {
        todo:[]
    }

    todoRef = React.createRef();

    handleNewTodo = (e) => {
        let todoClone = [...this.state.todo];
        let newTodo = this.todoRef.current.value;
        todoClone.push(newTodo);
        this.setState({todo : todoClone});
        
        e.preventDefault();
        this.todoRef.current.value = '';
    }

    render(){
        const {todo} = this.state;

        return(
            <div className="App App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <div>Todo List</div>
                <form onSubmit = {this.handleNewTodo}>
                    <label>New Todo <input ref={this.todoRef} /></label>
                </form>
                <ul>{todo.map((item,index) => <li key={index}>{item}</li>)}</ul>
            </div>

        )
    }
}