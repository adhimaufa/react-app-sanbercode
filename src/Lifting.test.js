import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import {CalculatorWeight} from './Components/pageLiftingSU';

test('input calc weight', () => {
  const component = renderer.create(
     <CalculatorWeight />
  );

  let tree = component.toJSON;
  expect(tree).toMatchSnapshot();

//   component.getInstance().handleGramChange();
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
});
