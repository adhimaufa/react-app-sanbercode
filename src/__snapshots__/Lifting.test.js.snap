// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input calc weight 1`] = `[Function]`;
